lib/call-cabal-project-to-nix.nix

```diff
@@ -212,13 +212,21 @@ let
                 (if inputMap.${repoData.url}.rev != repoData.ref
                   then throw "${inputMap.${repoData.url}.rev} may not match ${repoData.ref} for ${repoData.url} use \"${repoData.url}/${repoData.ref}\" as the inputMap key if ${repoData.ref} is a branch or tag that points to ${inputMap.${repoData.url}.rev}."
                   else inputMap.${repoData.url})
-            else if repoData.sha256 != null
+            else if repoData.sha256 != null && !repoData.is-private
               then fetchgit { inherit (repoData) url sha256; rev = repoData.ref; }
             else
-              let drv = builtins.fetchGit { inherit (repoData) url ref; };
-              in __trace "WARNING: No sha256 found for source-repository-package ${repoData.url} ${repoData.ref} download may fail in restricted mode (hydra)"
-                (__trace "Consider adding `--sha256: ${hashPath drv}` to the ${cabalProjectFileName} file or passing in a sha256map argument"
-                 drv);
+              let
+                # If the "tag" is a hash, we'll use 'rev', otherwise we'll use
+                # 'ref'. Using a commit hash as a 'ref' doesn't do what we want.
+                refOrRev = if builtins.match "[0-9a-f]*" (repoData.ref) != null
+                  then { rev = repoData.ref; }
+                  else { inherit (repoData) ref; };
+                drv = builtins.fetchGit ({ inherit (repoData) url; } // refOrRev);
```

Comment:
I think this pattern is also needlessly discarding information, too.

`{ inherit (repoData) url; } // refOrRev` - unless I'm mistaken, this means that we *only* will have a `{ url = _; "${ref or rev}": _ }` field. Which means we *can't* have both `rev` *and* `ref`, which is a problem I'm staring it.